include/llvm/Transforms/LiquidFixpointBuilder/VariablesEnvironmentImmutable.h

```diff
@@ -0,0 +1,70 @@
+#ifndef LLVM_TRANSFORMS_UTILS_LIQUID_VARIABLESENVIRONMENTIMMUTABLE_H
+#define LLVM_TRANSFORMS_UTILS_LIQUID_VARIABLESENVIRONMENTIMMUTABLE_H
+
+#include <map>
+#include <string>
+#include <set>
+#include <vector>
+#include "llvm/Transforms/LiquidFixpointBuilder/FixpointConstraintBuilder.h"
+#include "llvm/Transforms/LiquidFixpointBuilder/FixpointType.h"
+#include "llvm/Transforms/LiquidFixpointBuilder/FunctionBlockGraph.h"
+
+using namespace std::literals::string_literals;
+
+namespace liquid
+{
+  class VariablesEnvironmentImmutable
+  {
+    private:
+      std::string currentBlockName = ""s;
+      const FunctionBlockGraph& functionBlockGraph;
+
+      std::map<std::string, FixpointType> variableTypes;
+      std::map<std::string, std::map<std::string, std::string>> variablesMappingsPerBlock;
+
+      std::map<std::string, std::set<std::string>> variablesValuesPerBlock;
```

Comment:
change to vector

Reply:
I've changed this to a vector, and added a `RefinementUtils` function to get a set from a vector. As a result, I'm wondering if this should be changed in the first place ?